### PR TITLE
feat: embeds!

### PR DESCRIPTION
**Pull request by jennspencer**
![icn] | Fix RM-9017
:-------------------:|:----------:

## 🧰 Changes

embeds!!!

## 🧬 QA & Testing

~still need to `make updateSnapshot`~

[icn]: https://user-images.githubusercontent.com/886627/160426047-1bee9488-305a-4145-bb2b-09d8b757d38a.svg

### CODE REVIEW

**Comment by jennspencer** [comment]
@kellyjosephprice i dinked around a lot with types and tests here, which looks like it caused some conflicts 😂 i'll muddle through them but will defer to whatever patterns you're using

File: processor/transform/embeds.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+          data: {
+            hProperties: { title: node.children[0]?.value ?? node.url, url: node.url, provider: node.url },
+            hName: 'Embed',
+          },
+          position: node.position,
+          children: [],
+        };
+        parent = newNode;
+      } catch (e) {
+        console.log(e);
```
Comment (kellyjosephprice) [inline processor/transform/embeds.ts:21]: Did you mean to swallow the error?

File: types.d.ts [hunk 2]
```diff
@@ -40,6 +57,8 @@ declare module 'mdast' {
   interface BlockContentMap {
     [NodeTypes.callout]: Callout;
     [NodeTypes.codeTabs]: CodeTabs;
+    [NodeTypes.embed]: Embed;
```
Comment (kellyjosephprice) [inline types.d.ts:60]: I believe you want to add this to the `RootContentMap` too. I'm guessing it's probably the case that the link that the embed becomes is always wrapped in a paragraph, but that seems incorrect.

File: processor/transform/embeds.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import { visit } from 'unist-util-visit';
+
+import { NodeTypes } from '../../enums';
+
+const embedTransformer = () => {
+  return (tree: any) => {
+    visit(tree, 'link', (node, _, parent) => {
+      try {
+        if (parent.children.length > 1 || node.title !== '@embed') return;
```
Comment (kellyjosephprice) [inline processor/transform/embeds.ts:9]: Perhaps we should check if the parent is a paragraph, and completely replace it?

**Review by kellyjosephprice — CHANGES_REQUESTED** [review]
LGTM. The only one that really concerns me is the error.

File: processor/transform/embeds.ts (hunk 1 above)
Comment (jennspencer) [reply processor/transform/embeds.ts:21]: oh whoops

File: processor/transform/embeds.ts (hunk 3 above)
Comment (jennspencer) [reply processor/transform/embeds.ts:9]: yeah i should still be checking if the parent is a paragraph, i think i uh optimized that out at one point, heh. but i'm basically just reforming the node and replacing the parent with it--should i be splicing it in and out i or something nstead?

File: types.d.ts (hunk 2 above)
Comment (jennspencer) [reply types.d.ts:60]: ok cool, i wasn't 100% sure

**Comment by kellyjosephprice** [comment]
I think I fixed the syntax highlighter dynamic imports on the `beta` branch

**Comment by kellyjosephprice** [comment]
> @kellyjosephprice the `Code` component was/is working already, yeah? these errors are just on the test i edited so there's likely more tweaks it needs. the snapshot not matching is odd tho, wtf?
> 
> we can add a couple of todos to fix these guys, it's the last "convert component" ticket and i'd like to get crackin' on the editor-side component(s), whatchoo say?

Yea, it seems like there's something weird happening with the compilation step and the dynamic import?

Skip it!

**Comment by jennspencer** [comment]
> > @kellyjosephprice the `Code` component was/is working already, yeah? these errors are just on the test i edited so there's likely more tweaks it needs. the snapshot not matching is odd tho, wtf?
> > we can add a couple of todos to fix these guys, it's the last "convert component" ticket and i'd like to get crackin' on the editor-side component(s), whatchoo say?
> 
> Yea, it seems like there's something weird happening with the compilation step and the dynamic import?
> 
> Skip it!

YOLO

**Comment by rafegoldberg** [comment]
## This PR was released!

🚀 Changes included in [<b>v6.75.0-beta.34</b>](/readmeio/markdown/releases/tag/v6.75.0-beta.34)
<br>

**Pull request merged**